src/utils/ProteomicsLFQ.cpp

```diff
@@ -0,0 +1,1598 @@
+-ids ../share/OpenMS/examples/FRACTIONS/BSA1_F1.idXML ../share/OpenMS/examples/FRACTIONS/BSA1_F2.idXML ../share/OpenMS/examples/FRACTIONS/BSA2_F1.idXML ../share/OpenMS/examples/FRACTIONS/BSA2_F2.idXML ../share/OpenMS/examples/FRACTIONS/BSA3_F1.idXML ../share/OpenMS/examples/FRACTIONS/BSA3_F2.idXML \
+-design ../share/OpenMS/examples/FRACTIONS/BSA_design.tsv \
+-Alignment:max_rt_shift 0 \
+-fasta ../share/OpenMS/examples/TOPPAS/data/BSA_Identification/18Protein_SoCe_Tr_detergents_trace_target_decoy.fasta \
+-out BSA.mzTab -threads 4 -debug 667 
+
+
+Potential script to perform the search
+# perform search, score calibration, and PSM-level q-value/PEP estimation
+for f in ${DATA_PATH}*.mzML; do
+  echo $f
+  fn=${f%.mzML} # filename and path without mzML extension
+  # search with default fixed and variable mods
+  MSGFPlusAdapter -in ${f} -out ${fn}.idXML -database ${DATA_PATH}/iPRG2015_decoy.fasta -executable ${MSGF_PATH} -max_precursor_charge 5 -threads 10 
+  # annotate target/decoy and protein links
+  PeptideIndexer -fasta ${DATA_PATH}/iPRG2015_decoy.fasta  -in ${fn}.idXML -out ${fn}.idXML -enzyme:specificity none
+  # run percolator so we get well calibrated PEPs and q-values
+  PSMFeatureExtractor -in ${fn}.idXML -out ${fn}.idXML 
+  PercolatorAdapter -in ${fn}.idXML -out ${fn}.idXML -percolator_executable percolator -post-processing-tdc -subset-max-train 100000 
+  FalseDiscoveryRate -in ${fn}.idXML -out ${fn}.idXML -algorithm:add_decoy_peptides -algorithm:add_decoy_proteins 
+  # pre-filter to 5% PSM-level FDR to reduce data
+  IDFilter -in ${fn}.idXML -out ${fn}.idXML -score:pep 0.05 
+  # switch to PEP score
+  IDScoreSwitcher -in ${fn}.idXML -out ${fn}.idXML -old_score q-value -new_score MS:1001493 -new_score_orientation lower_better -new_score "Posterior Error Probability" 
+done
+ **/
+
+class UTILProteomicsLFQ :
+  public TOPPBase
+{
+public:
+  UTILProteomicsLFQ() :
+    TOPPBase("ProteomicsLFQ", "A standard proteomics LFQ pipeline.", false)
+  {
+  }
+
+protected:
+
+  void registerOptionsAndFlags_() override
+  {
+    registerInputFileList_("in", "<file list>", StringList(), "Input files");
+    setValidFormats_("in", ListUtils::create<String>("mzML"));
+    registerInputFileList_("ids", "<file list>", StringList(), 
+      "Identifications filtered at PSM level (e.g., q-vaue < 0.01)."
+      "And annotated with PEP as main score.\n"
+      "We suggest using:\n"
+      "1. PercolatorAdapter tool (score_type = 'q-value', -post-processing-tdc)\n"
+      "2. FalseDiscoveryRate (FDR:PSM = 0.01)\n"
+      "3. IDScoreSwitcher (-old_score q-value -new_score MS:1001493 -new_score_orientation lower_better -new_score_type)\n"
+      "To obtain well calibrated PEPs and an inital reduction of PSMs\n"
```

Comment:
@timosachsenberg why not have the LFQ workflow do this automatically? does the user have to run this sepearatley?

Reply:
Yes but we could move it into the tool. Will have a strong dependency to Percolator though.

Reply:
Not needed anymore. You can select PEP in PercolatorAdapter and it fills the score_type with "PosteriorErrorProbability"

Reply:
ah you also want to allow filtering. sorry. hmm

Reply:
Still I would remove it from the advice and do it in the tool.

Reply:
> Yes but we could move it into the tool. Will have a strong dependency to Percolator though.

I suggest that the user can supply their own ID files if they want to or they can only provide mzML and fasta and then the tool will do the search for them. This would provide a "MaxQuant" like feeling for the user - maybe only pick a single search engine for this, no need to support many search engines here.